src/view/xwayland.cpp

```diff
@@ -305,9 +306,24 @@ class wayfire_xwayland_view : public wayfire_xwayland_view_base
 
         on_request_move.set_callback([&] (void*) { move_request(); });
         on_request_resize.set_callback([&] (void*) { resize_request(); });
+
+        on_request_activate.set_callback([&] (void*) {
+            if (!this->activated)
+            {
+                view_self_request_focus_signal data;
+                data.view = this;
+                this->get_output()->emit_signal("view-self-request-focus", &data);
+            }
+        });
+        on_request_minimize.set_callback([&] (void*) {
+            minimize_request(xw->minimized);
+        });
         on_request_maximize.set_callback([&] (void*) {
-            tile_request((xw->maximized_horz && xw->maximized_vert) ?
-                wf::TILED_EDGES_ALL : 0);
+            if (xw->maximized_horz && xw->maximized_vert)
```

Comment:
wait why the change? It doesn't do anything ...

Reply:
I mean it is the same as before

Reply:
You looked at the newer commits which almost all say test,
because of the switcher plugin :)

There is one other issue I have been looking into:
https://github.com/ValveSoftware/steam-for-linux/issues/3792

this change was basically a result though that these are different properties and possibly at some point this should be changed further to actually make partial maximize recognized (i.e horz / vert)

Reply:
Just look at the code before and now, you have not changed ANYTHING.

Reply:
Well, except that before it was using ternary operator and now it uses if/else ..

Reply:
If you implement at one point vert/horz maximization in this function it will have to change from the ternary operator proably. I wanted to implement this but wlroots has no signal for vert/horz tile changes yet. I can revert this.

Reply:
I'd keep it as it was before, until the moment we implement separate maximization.

Reply:
Done